src/msgs/data_requests/execute/commit_result.rs

```diff
@@ -10,6 +10,51 @@ pub struct Execute {
     pub proof:      String,
 }
 
+impl SignSelf for Execute {
+    type Extra = u64;
+
+    fn proof(&self) -> Result<Vec<u8>> {
+        Ok(hex::decode(&self.proof)?)
+    }
+
+    fn msg_hash(&self, chain_id: &str, contract_addr: &str, dr_height: Self::Extra) -> Result<Hash> {
+        Ok(hash([
+            "commit_data_result".as_bytes(),
+            self.dr_id.as_bytes(),
+            &dr_height.to_be_bytes(),
+            self.commitment.as_bytes(),
+            chain_id.as_bytes(),
+            contract_addr.as_bytes(),
+        ]))
+    }
+}
+
+impl Execute {
+    pub fn new(
+        dr_id: String,
+        commitment: String,
+        public_key: String,
+        signing_key: &[u8],
+        chain_id: &str,
+        contract_addr: &str,
+        dr_height: u64,
```

Comment:
Should not all params be string slices?

```suggestion
        dr_id: &str,
        commitment: &str,
        public_key: &str,
        signing_key: &[u8],
        chain_id: &str,
        contract_addr: &str,
        dr_height: u64,
```

Same applies to places.

Reply:
No. The ones that are String are put into a struct. So if we passed a &str we would allocated it as a String anyways.

I suppose that's fine if we are sure we aren't passing it already allocated strings by reference to avoid allocating another instance of that same string.

Reply:
It is more ergonomic also because in some parts of the code the moved value is later used (we can always add a `clone()`).

Reply:
Yes I would clone where you call new if needed.